controllers/secret_controller.go

```diff
@@ -149,6 +155,19 @@ func (r *SecretReconciler) Reconcile(ctx context.Context,
 			result.Requeue, reconcileErr)
 	}()
 
+	if request.NamespacedName.Name == tls.SecretName {
+		tlsSecret := &core.Secret{}
+		err := r.client.Get(context.TODO(), kubeTypes.NamespacedName{Name: tls.SecretName,
+			Namespace: r.watchNamespace}, tlsSecret)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("unable to get secret %s: %w", r.watchNamespace, err)
+		}
+		err = metadata.ApplySecretTlsAnnotation(ctx, r.client, tlsSecret)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("unable to apply TLS annotation %s: %w", r.watchNamespace, err)
+		}
```

Comment:
You need to return at the end of this segment.

Reply:
In addition, I suggest pulling this out into a helper func so this call site could become something like 
`return reconcile.Result{}, r.reconcileTLSSecret()`